assets/js/daily-form.js

```diff
@@ -0,0 +1,63 @@
+export default class DailyForm {
+  static init () {
+    const addGoalButton = document.getElementById('add-goal-btn')
+    if (addGoalButton) {
+      addGoalButton.addEventListener('click', (event) => {
+        this.addGoal()
+      }, false)
+    }
+  }
+
+  static addGoal () {
+    const goalsDiv = document.querySelector('.goals-form')
+    const goalsCount = goalsDiv.querySelectorAll('.form-control').length
+    const newInput = this.createGoalForm(goalsCount)
+
+    goalsDiv.appendChild(newInput)
+  }
+
+  static createGoalForm (goalsCount) {
```

Comment:
I was thinking it would be nice to leave the rendering to Phoenix as a partial and deliver the content over a channel. Selfishly I need to test that out for my day job, so I hadn't thought of a js approach. I'm happy with this unless you're also curious about that approach.

Reply:
100% that sounds like a way better way to go 👍